### ISSUE TITLE
Highlighting: expose merge_contiguos plain highlighter options

### ISSUE THREAD

**Issue by javanna** [elastic/elasticsearch#12376]
Some users end up using the fast vector highlighter because it is able to merge contiguos fragments and add the highlighting tags only once e.g. `<em>onetwothree</em>` rather than once per match e.g. `<em>one</em><em>two</em><em>three</em>`. I think this is not a good reason to use the fast vector highlighter, and I see that the lucene plain highlighter supports merging contiguous fragments but we don't expose this option to the users and always pass in `false` to lucene. I wonder if it would make sense to expose this setting, which would be read only by the plain highlighter and ignored by other implementations, so that the plain highlighter is able to merge contiguos fragments if requested.

**Comment by nik9000** [comment]
In theory this sounds good to me. This is exactly the kind of thing that I'd just fix on the client side if I didn't like it though. But if its simple to turn on then why not?

**Comment by javanna** [comment]
We discussed this during fix-it friday, I totally misunderstood the `mergeContiguous` option, which merges contiguous fragments rather than contiguous tokens. This might be a useful option but given all the options that we already have and which work only for some of the highlighters (this one would work only for the plain one), I would not expose it for now. We will also start some more general discussion around highlighting and how we want to improve things there post 2.0. Closing.
